consul/client.go

```diff
@@ -182,10 +182,41 @@ func isKeyIndexMismatchErr(err error) bool {
 func newKvClient(machine string, p connectionParams) (*api.Config, *api.Client, error) {
 	logrus.Infof("consul: connecting to %v", machine)
 	config := api.DefaultConfig()
-	config.HttpClient = http.DefaultClient
-	config.Address = machine
-	config.Scheme = "http"
-	config.Token = p.options[kvdb.ACLTokenKey]
+
+	// check if TLS is required
+	caFile, ok1 := p.options[kvdb.CAFileKey]
+	certFile, ok2 := p.options[kvdb.CertFileKey]
+	certKeyFile, ok3 := p.options[kvdb.CertKeyFileKey]
+	caAuthAddress := p.options[kvdb.CAAuthAddress]
+	insecureSkipVerify := strings.ToLower(p.options[kvdb.InsecureSkipVerify]) == "true"
+	if ok1 && ok2 && ok3 {
```

Comment:
In etcd there is a way to setup where only CAFile is required and certFile and certKey is not required. Can consul be setup in such a way? If so we cannot have a hard check on all the ok's here.

Reply:
i am not sure. i do have a consul cluster with self signed TLS. i could give it a try there.